### PR TITLE
Update RegionSelect to be one file, add test

### PR DESCRIPTION
**Pull request by bryce-mcmath**
### Please check if the PR fulfills these requirements:

- [ ] After these changes, the app was run and still works as expected
- [x] Tests for these changes were added (if applicable)
- [ ] All existing unit tests were run and still pass

### Please specify the type of change this PR introduces (Bug fix, feature addition, content update, chore, etc.):
Chore (vue3 upgrade)

### Additional Notes:
Quite a few changes, take your time

### CODE REVIEW

File: src/components/RegionSelect.vue [hunk 1]
```diff
@@ -78,11 +92,100 @@ export default {
...
+  }),
+  mounted() {
+    if (this.country) {
+      this.getRegionWithCountry()
+    } else {
+      let findRegion = ''
+      if (this.countryName) {
+        findRegion = this.defaultRegion ? this.defaultRegion : 'United States'
+      } else {
+        findRegion = this.defaultRegion ? this.defaultRegion : 'US'
```
Comment (harrymaynard-maximus) [inline src/components/RegionSelect.vue:121]: I would think the fallback region would be Canada, no?

**Review by harrymaynard-maximus — CHANGES_REQUESTED** [review]

File: src/components/RegionSelect.vue (hunk 1 above)
Comment (bryce-mcmath) [reply src/components/RegionSelect.vue]: Ah good catch, that was leftover from VueRegionSelect

**Review by harrymaynard-maximus — APPROVED** [review]
Looks good.

**Review by jlangy — APPROVED** [review]
Looks good!

**Pull request merged**